controllers/storagecluster/cephtoolbox.go

```diff
@@ -117,21 +119,34 @@ func removeDuplicateTolerations(tolerations []corev1.Toleration) []corev1.Tolera
 	return list
 }
 
-func getMultusPublicNetwork(sc *ocsv1.StorageCluster) (string, error) {
+func (r *StorageClusterReconciler) getMultusPublicNetwork(sc *ocsv1.StorageCluster) (string, error) {
 	err := validateMultusSelectors(sc.Spec.Network.Selectors)
 	if err != nil {
 		return "", err
 	}
 
+	if sc.Spec.Network.Selectors["public"] == "" {
+		return "", nil
+	}
+
+	multusNetName := sc.Spec.Network.Selectors["public"]
+	multusNetNamespacedName := strings.Split(multusNetName, "/")
+	if len(multusNetNamespacedName) == 2 {
+		nadNS, nadName := multusNetNamespacedName[0], multusNetNamespacedName[1]
+		foundNAD := &nadv1.NetworkAttachmentDefinition{}
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: nadName, Namespace: nadNS}, foundNAD)
+		if err != nil {
+			return "", err
+		}
+		return multusNetName, nil
+	} else if len(multusNetNamespacedName) > 2 {
+		return "", fmt.Errorf("public selector value: %s, is not as per specification", multusNetName)
+	}
+
 	multusNetNs, err := util.GetOperatorNamespace()
 	if err != nil {
 		return "", err
 	}
 
-	if sc.Spec.Network.Selectors["public"] != "" {
-		multusNetName := sc.Spec.Network.Selectors["public"]
-		return fmt.Sprintf("%s/%s", multusNetNs, multusNetName), nil
-	}
-
-	return "", nil
+	return fmt.Sprintf("%s/%s", multusNetNs, multusNetName), nil
```

Comment:
As we are doing the get on the one if it is another namespace should not we do get even if it is in the same namespace?

Reply:
Ah, right. It would be good to test it for the operator ns as well.
Updated the code accordingly, PTAL.